sklearn/semi_supervised/_self_training.py

```diff
@@ -169,25 +192,58 @@ class SelfTrainingClassifier(
 
     def __init__(
         self,
-        base_estimator,
+        estimator=None,
+        base_estimator="deprecated",
         threshold=0.75,
         criterion="threshold",
         k_best=10,
         max_iter=10,
         verbose=False,
     ):
-        self.base_estimator = base_estimator
+        self.estimator = estimator
         self.threshold = threshold
         self.criterion = criterion
         self.k_best = k_best
         self.max_iter = max_iter
         self.verbose = verbose
 
+        # TODO(1.8) remove
+        self.base_estimator = base_estimator
+
+    def get_estimator(self):
+        """Get the estimator.
+
+        Returns
+        -------
+        estimator_ : estimator object
+            The cloned estimator object.
+        """
+        # TODO(1.8): remove and only keep clone(self.estimator)
+        if self.estimator is None and self.base_estimator != "deprecated":
+            estimator_ = clone(self.base_estimator)
+
+            warn(
+                (
+                    "`base_estimator` has been deprecated in 1.6 and will be removed"
+                    " in 1.8. Please use `estimator` instead."
+                ),
+                FutureWarning,
+            )
+        # TODO(1.8) remove
+        elif self.estimator is None and self.base_estimator == "deprecated":
+            raise ValueError(
+                "You must pass an estimator to SelfTrainingClassifier."
+                " Use `estimator`."
+            )
+        else:
+            estimator_ = clone(self.estimator)
+        return estimator_
```

Comment:
still missing the case where both `estimator` and `base_estimator` are passed, in which case we need to raise

Reply:
[42dade6](https://github.com/scikit-learn/scikit-learn/pull/28494/commits/42dade665563cafb7076bbe12e987fbed3029abf)